pantheon/src/main/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommand.java

```diff
@@ -0,0 +1,319 @@
+      LOG.info("Importing public keys from configuration.");
+      JsonArray keys = nodesConfig.getJsonArray("keys");
+      keys.stream().forEach(this::importPublicKey);
+    }
+
+    /**
+     * Imports a single public key.
+     *
+     * @param publicKeyObject The public key.
+     */
+    private void importPublicKey(final Object publicKeyObject) {
+      try {
+        final SECP256K1.PublicKey publicKey =
+            SECP256K1.PublicKey.create(BytesValue.fromHexString((String) publicKeyObject));
+        writeKeypair(publicKey, null);
+        LOG.info("Public key imported from configuration.({})", publicKey.toString());
+      } catch (IOException e) {
+        LOG.error("An error occurred while trying to import node public key.", e);
+      }
+    }
+
+    /** Generates nodes keypairs. */
+    private void generateNodesKeys() {
+      final int nodesCount = nodesConfig.getInteger("count");
+      LOG.info("Generating {} nodes keys.", nodesCount);
+      IntStream.range(0, nodesCount).forEach(this::generateNodeKeypair);
+    }
+
+    /**
+     * Generate a keypair for a node.
+     *
+     * @param node The number of the node.
+     */
+    private void generateNodeKeypair(final int node) {
+      try {
+        LOG.info("Generating keypair for node {}.", node);
+        final SECP256K1.KeyPair keyPair = SECP256K1.KeyPair.generate();
+        writeKeypair(keyPair.getPublicKey(), keyPair.getPrivateKey());
+
+      } catch (IOException e) {
+        LOG.error("An error occurred while trying to generate node keypair.", e);
+      }
+    }
+
+    /**
+     * Writes public and private keys in separate files. Both are written in the same directory
+     * named with the address derived from the public key.
+     *
+     * @param publicKey The public key.
+     * @param privateKey The private key. No file is created if privateKey is NULL.
+     * @throws IOException
+     */
+    private void writeKeypair(
+        final SECP256K1.PublicKey publicKey, final SECP256K1.PrivateKey privateKey)
+        throws IOException {
+      final Address nodeAddress = Util.publicKeyToAddress(publicKey);
+      addressesForGenesisExtraData.add(nodeAddress.toUnprefixedString());
+      final Path nodeDirectoryPath = keysDirectory.resolve(nodeAddress.toString());
+      Files.createDirectory(nodeDirectoryPath);
+      createFileAndWrite(nodeDirectoryPath, publicKeyFileName, publicKey.toString());
+      if (privateKey != null) {
+        createFileAndWrite(nodeDirectoryPath, privateKeyFileName, privateKey.toString());
+      }
+    }
+
+    /**
+     * Computes RLP encoded exta data from pre filled list of addresses.
+     *
+     * @throws IOException
+     */
+    private void processExtraData() throws IOException {
+      LOG.info("Generating genesis extra data.");
+      final String jsonAddresses = Json.encodePrettily(addressesForGenesisExtraData);
+      final String extraData = IbftExtraData.fromJsonAddresses(jsonAddresses).encode().toString();
```

Comment:
Since we're assuming a consensus algorithm of ibft, we should also validate that the genesis config has the "ibft2" field.  

(Optional) Alternatively as @rain-on suggested, you could check for the [consensus field](https://github.com/PegaSysEng/pantheon/blob/28437b0c5e69637a0e081a4a4df7fcf14f5148bc/config/src/main/java/tech/pegasys/pantheon/config/JsonGenesisConfigOptions.java#L28-L31) and use the consensus-specific extradata utility here.

Reply:
Done.